pinot-core/src/test/java/com/linkedin/pinot/core/data/readers/RecordReaderTest.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.data.readers;
+
+import com.linkedin.pinot.common.data.FieldSpec;
+import com.linkedin.pinot.common.data.Schema;
+import com.linkedin.pinot.core.data.GenericRow;
+import java.io.File;
+import java.io.FileWriter;
+import org.apache.commons.csv.CSVFormat;
+import org.apache.commons.csv.CSVPrinter;
+import org.apache.commons.io.FileUtils;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
+public class RecordReaderTest {
+  private static final File TEMP_DIR = new File(FileUtils.getTempDirectory(), "RecordReaderTest");
+  private static final File CSV_DATA_FILE = new File(TEMP_DIR, "data.csv");
+  private static final File JSON_DATA_FILE = new File(TEMP_DIR, "data.json");
+  private static final Schema SCHEMA = new Schema.SchemaBuilder().addMetric("INT_SV", FieldSpec.DataType.INT)
+      .addMultiValueDimension("INT_MV", FieldSpec.DataType.INT, -1)
+      .build();
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    FileUtils.forceMkdir(TEMP_DIR);
+
+    // Create CSV data file
+    try (FileWriter fileWriter = new FileWriter(CSV_DATA_FILE);
+        CSVPrinter csvPrinter = new CSVPrinter(fileWriter, CSVFormat.DEFAULT.withHeader("INT_SV", "INT_MV"))) {
```

Comment:
can we create a two-dimensional array to represent the data and initialize that in the setup and that can be used in other places instead of hard-coding the values in multiple places

Reply:
Done